initrd/bin/seal-hotpkey

```diff
@@ -121,38 +130,50 @@ if [ "$((now_date - gpg_key_create_time))" -gt "$month_secs" ]; then
+	echo "Trying $prompt_message PIN ($admin_pin) to seal HOTP secret on $HOTPKEY_BRANDING..."
+	#if we deal with the nk3, say to the user that touch will be required
+	if lsusb | grep -q "20a0:42b2"; then
+		warn "Nitrokey 3 requires physical presence : touch the dongle when prompted"
+		echo
+	fi
+	#TODO: silence the output of hotp_initialize once https://github.com/Nitrokey/nitrokey-hotp-verification/issues/41 is fixed
+	#hotp_initialize "$admin_pin" $HOTP_SECRET $counter_value "$HOTPKEY_BRANDING" >/dev/null 2>&1
+	hotp_initialize "$admin_pin" $HOTP_SECRET $counter_value "$HOTPKEY_BRANDING"
 	admin_pin_status="$?"
 fi
 
 if [ "$admin_pin_status" -ne 0 ]; then
-  # prompt user for PIN and retry
-  echo ""
-  read -s -p "Enter your $HOTPKEY_BRANDING Admin PIN: " admin_pin
-  echo -e "\n"
-  
-  hotp_initialize "$admin_pin" $HOTP_SECRET $counter_value "$HOTPKEY_BRANDING"
-  if [ $? -ne 0 ]; then
-    echo -e "\n"
-    read -s -p "Error setting HOTP secret, re-enter Admin PIN and try again: " admin_pin
-    echo -e "\n"
-    if ! hotp_initialize "$admin_pin" $HOTP_SECRET $counter_value "$HOTPKEY_BRANDING" ; then
-      # don't leak key on failure
-      shred -n 10 -z -u "$HOTP_SECRET" 2> /dev/null
-      if [ "$HOTPKEY_BRANDING" == "Nitrokey" ]; then
-	fatal_error "Setting HOTP secret failed, to reset nitrokey pin use: nitropy nk3 secrets reset or the Nitrokey App 2"
-      else
-	fatal_error "Setting HOTP secret failed"
-      fi
-    fi
-  fi
-else 
-  # remind user to change admin password
-  echo -e "\nWARNING: default admin PIN detected: please change this as soon as possible."
+
+	# prompt user for PIN and retry
+	echo ""
+	read -s -p "Enter your $HOTPKEY_BRANDING $prompt_message PIN: " admin_pin
+	echo -e "\n"
+
+	hotp_initialize "$admin_pin" $HOTP_SECRET $counter_value "$HOTPKEY_BRANDING"
+	if [ $? -ne 0 ]; then
+		echo -e "\n"
+		read -s -p "Error setting HOTP secret, re-enter $prompt_message PIN and try again: " admin_pin
+		echo -e "\n"
+		if ! hotp_initialize "$admin_pin" $HOTP_SECRET $counter_value "$HOTPKEY_BRANDING"; then
+			# don't leak key on failure
+			shred -n 10 -z -u "$HOTP_SECRET" 2>/dev/null
+			if [ "$HOTPKEY_BRANDING" == "Nitrokey" ]; then
+				fatal_error "Setting HOTP secret failed, to reset $prompt_message PIN, redo Re-Ownership procedure, use the Nitrokey App 2 or contact Nitrokey support"
+			else
+				fatal_error "Setting HOTP secret failed"
+			fi
+		fi
+	fi
+else
+	# remind user to change admin password
+	warn "Weak OEM default PINs are under use to enforce remote attestation/encryption/signature operations"
+	warn "$CONFIG_BRAND_NAME security is compromised until the ownership of this device is re-established by changing secrets by non-default values"
+	warn "You must change current default secrets through 'Options -> OEM Factory Reset/Re-Ownership' menu and not accept the default options"
+	warn "You will be asked to answer a questionnaire to re-own your device and USB security dongles with new secrets"
```

Comment:
IMO, the amount of text in this warning will cause fewer people to read it, not more.  (I am guilty of this.)  I know I have read research on this, sadly I don't have enough time today to look it up.  Above a sentence or two, users very frequently ignore the message entirely.

It _is_ a great improvement to suggest where to go rather than leaving users totally confused.  There is no sense re-explaining what OEM factory reset will explain anyway though.

```suggestion
	# remind user to change admin password
	warn "Default admin PIN detected.  Please change this as soon as possible with Options > OEM Factory Reset / Re-Ownership"
```

Reply:
Fixed under [94dd788](https://github.com/linuxboot/heads/pull/1875/commits/94dd788249c65b5b4667e4cddecaa4f53e46e9ca) which shows Secrets app PIN/ GPG Admin PIN depending if nk3/<nk3 (tested on 0.10 fw librem key: ok with branding info ok as well.